gateway/src/main/kotlin/retry/LinearRetry.kt

```diff
@@ -42,10 +43,13 @@ public class LinearRetry(
     override suspend fun retry() {
         if (!hasNext) error("max retries exceeded")
 
-        tries.incrementAndGet()
-        var diff = (maxBackoff - firstBackoff).inWholeMilliseconds / maxTries
-        diff *= tries.value
-        linearRetryLogger.trace { "retry attempt ${tries.value}, delaying for $diff ms" }
+        // tries/maxTries ratio * (backOffDiff) = retryProgress
+        val ratio = tries.getAndIncrement() / (maxTries - 1).toDouble()
+        val retryProgress =
+            (ratio * (maxBackoff - firstBackoff).inWholeMilliseconds).toLong().milliseconds
```

Comment:
If you import [`kotlin.time.times`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.time/times.html), you can also avoid the conversion here.

Reply:
oh cool, intellij didn't show that so I assumed it didn't exist